crates/solidity/outputs/rust/lib/src/generated/language.rs

```diff
@@ -170,13 +170,35 @@ pub struct Language {
     pub(crate) version_is_equal_to_or_greater_than_0_8_18: bool,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Invalid version '{0}'.")]
+    InvalidVersion(Version),
+}
+
 impl Language {
-    pub fn new(version: Version) -> Self {
-        Self {
-            version_is_equal_to_or_greater_than_0_6_0: Version::parse("0.6.0").unwrap() <= version,
-            version_is_equal_to_or_greater_than_0_8_18: Version::parse("0.8.18").unwrap()
-                <= version,
-            version,
+    pub fn new(version: Version) -> Result<Self, Error> {
+        static VERSIONS: &'static [&'static str] = &[
+            "0.4.11", "0.4.12", "0.4.13", "0.4.14", "0.4.15", "0.4.16", "0.4.17", "0.4.18",
+            "0.4.19", "0.4.20", "0.4.21", "0.4.22", "0.4.23", "0.4.24", "0.4.25", "0.4.26",
+            "0.5.0", "0.5.1", "0.5.2", "0.5.3", "0.5.4", "0.5.5", "0.5.6", "0.5.7", "0.5.8",
+            "0.5.9", "0.5.10", "0.5.11", "0.5.12", "0.5.13", "0.5.14", "0.5.15", "0.5.16",
+            "0.5.17", "0.6.0", "0.6.1", "0.6.2", "0.6.3", "0.6.4", "0.6.5", "0.6.6", "0.6.7",
+            "0.6.8", "0.6.9", "0.6.10", "0.6.11", "0.6.12", "0.7.0", "0.7.1", "0.7.2", "0.7.3",
+            "0.7.4", "0.7.5", "0.7.6", "0.8.0", "0.8.1", "0.8.2", "0.8.3", "0.8.4", "0.8.5",
+            "0.8.6", "0.8.7", "0.8.8", "0.8.9", "0.8.10", "0.8.11", "0.8.12", "0.8.13", "0.8.14",
+            "0.8.15", "0.8.16", "0.8.17", "0.8.18",
+        ];
+        if VERSIONS.contains(&version.to_string().as_str()) {
```

Comment:
nit suggestion, as the method call threw me off. I expected it to do something more complex than just returning a reference, or possibly panic if the conversion failed.

```suggestion
        if VERSIONS.contains(&&version.to_string()) {
```

Reply:
> The problem for me is that the two & have very different functions.

Valid point!